macos_icon: Fix skewed logo for Spotlight/Finder

pournaki opened the pull request:
macOS icons are skewed for sizes 16x16, 32x32, 48x48 (see issue #1316 ).

---

<!--
Remove the fields that are not appropriate
Please include:
-->

**What's this PR do?**
As previously mentioned in Issue #1316, the app icon in macOS looks skewed in Spotlight, Finder. This PR fixes this look.

**Any background context you want to provide?**
I created a new `icon.icns` based on the icons in `build/icons` using `iconutil -c icns`.

**Screenshots**
Before: 
![Screenshot 2024-09-18 at 10 45 34](https://github.com/user-attachments/assets/9d5c452f-f7a7-4c80-b509-fa2685aff103)
After:
![image](https://github.com/user-attachments/assets/2ad0b3a4-3a56-4cbc-ae22-8e259cf60055)

**You have tested this PR on:**
- [ ] Windows
- [ ] Linux/Ubuntu
- [x] macOS (14.6.1)

pournaki committed:
macos_icon: Fix skewed logo for Spotlight/Finder

macOS icons were skewed for sizes 16x16, 32x32, 48x48 (see issue #1316 ), affecting their display in Spotlight and Finder.

A fixed `icon.icns` was built based on the icons in `build/icons` using the macOS-internal command `iconutil -c icns`.

alya commented:
Thanks for the fix! Could you please update your commit message to match the [commit style guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages)?

pournaki commented:
Sure, does it comply now?

alya commented:
@andersk are you up for reviewing this one?

pournaki commented:
I made a mistake building the new logo. I took the files from `build/icons`, but they are missing a crucial margin. Without it, the logo looks too big in the app switcher. I am attaching a screenshot that compares a logo file extracted from `icon.icns` (left) to the same size one from `build/icons` (right). 

<img width="1707" alt="Screenshot 2024-09-23 at 14 52 00" src="https://github.com/user-attachments/assets/79c97b4b-a68f-433e-942d-486e30634949">

I have created a new `icon.icns` with the correct margins by resizing the 512px logo to the faulty sizes using 

```
$ sips -Z <SIZE> icon_512x512.png --out icon_<SIZE>x<SIZE>.png
```

Is there a way to update this pull request with the new file? Or should I close it and make a new one?

pournaki commented:
There is still a (probably) unwanted faint white border around the icon, visible in the Finder (pic1) and Spotlight (pic2). I don't think this is an artifact of the resizing operation.

![image](https://github.com/user-attachments/assets/c5b54d49-8287-4e3a-b95b-b08503bd890f)
![image](https://github.com/user-attachments/assets/e7e43786-8347-4829-bd65-f0c66120c8dc)

This line does not appear in the app switcher.

alya commented:
Yes, you can push to this PR. https://zulip.readthedocs.io/en/latest/git/using.html might help.

pournaki committed:
macos_icon: Revert to original icon margin

The previously generated `icon.icns` that fixed the skewed logos for small sizes was missing the necessary margin.
